nf_deployments/v1alpha1/nf_deployment_types.go

```diff
@@ -0,0 +1,201 @@
+	// dnns defines the dnns assocated with the network instance
+	// +optional
+	Dnns []*Dnn `json:"dnns,omitempty" yaml:"dnns,omitempty"`
+	// bgp defines the BGP configuration associated with the network instance
+	// +optional
+	Bgp *BGPConfig `json:"bgp,omitempty" yaml:"bgp,omitempty"`
+}
+
+// A Dnn defines the Data Network name defined by 3GPP
+type PeerConfig struct {
+	// Name defines the name of the data network
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// IPv4Config defines the ipv4 configuration of the peer
+	// +optional
+	IPv4Config *IPv4Config `json:"ipv4,omitempty" yaml:"ipv4,omitempty"`
+	// IPv6Config defines the ipv6 configuration of the peer
+	// +optional
+	IPv6Config *IPv6Config `json:"ipv6,omitempty" yaml:"ipv6,omitempty"`
+}
+
+// A Dnn defines the Data Network name defined by 3GPP
+type Dnn struct {
+	// Name defines the name of the data network
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// UEIPAddressPool defines the list of address pools associated with the dnn
+	// +optional
+	UEIPAddressPool []*IPAddressPool `json:"pool,omitempty" yaml:"pool,omitempty"`
+}
+
+type IPAddressPool struct {
+	// Prefix defines the ip cidr in prefix notation. It is defines as a subnet
+	// +kubebuilder:validation:Pattern=`(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])/(([0-9])|([1-2][0-9])|(3[0-2]))|((:|[0-9a-fA-F]{0,4}):)([0-9a-fA-F]{0,4}:){0,5}((([0-9a-fA-F]{0,4}:)?(:|[0-9a-fA-F]{0,4}))|(((25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])))(/(([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))`
+	Prefix string `json:"prefix" yaml:"prefix"`
+}
+
+// BgpConfig specifies parameters for BGP related configuration for UPF and SMF
+type BGPConfig struct {
+	// RouterID defines the router ID of the bgp process
+	RouterID string `json:"routerID" yaml:"prefix"`
+	// ASNumber defines the AS number of the bgp process
+	ASNumber int `json:"autonomousSystem" yaml:"autonomousSystem"`
+	// BGPNeigbors defines the configuration of the BGP neighbor
+	BGPNeigbors []*BGPNeighbor `json:"neighbors" yaml:"neighbors"`
+}
+
+type BGPNeighbor struct {
+	// Address defines the IPv4 or IPv6 address of the BGP neighbor
+	Address string `json:"address" yaml:"address"`
+	// BGP interface name, MUST match the one use in InterfaceConfig
+	Name string `json:"Name,omitempty"`
+	// PeerAS defines the AS number of the bgp peer
+	PeerAS int `json:"peerAS" yaml:"autonomousSystem"`
+}
+
+// NFDeploymentStatus defines the observed state of nf deployment
+type NFDeploymentStatus struct {
+	// The generation observed by the deployment controller.
+	ObservedGeneration int32 `json:"observedGeneration"`
+	// Current service state of the NF.
+	Conditions []Condition `json:"conditions,omitempty"`
+}
+
+type Condition struct {
```

Comment:
Is this identical to the standard meta conditions? If not, why not? If so, why include it here?

Reply:
Sorry for the late reply. Yes, if there is no difference between the two, I agree that we should just go with the condition definition from metav1, no need to define our own unless we have a clear use case for the condition. Thanks

Reply:
fixed